### ISSUE TITLE
Resolving incidents with error type TASK_LISTENER_NO_RETRIES

### ISSUE THREAD

**Issue by vitorwtavares** [camunda/camunda#23946]
Part of #21974 epic.

### Description
Resolving incidents with error type `TASK_LISTENER_NO_RETRIES` should update the retries and then retry the job.
The implementation should follow the same approach as #23383.

TODO

- [x] Add TASK_LISTENER_NO_RETRIES to [ErrorType enum](https://github.com/camunda/camunda/blob/main/webapps-schema/src/main/java/io/camunda/webapps/schema/entities/operate/ErrorType.java)
- [x] Expand tests to cover Task Listeners (see #23383)

### Acceptance Criteria

- [x] Failed Task Listeners with no retries left can be resolved via Operate
- [x] integration tests have been extended to also cover Task Listeners

#### Manual testing steps:
Note: Zeebe SNAPSHOT version required to support task listeners!

1. Run Zeebe, Operate and Tasklist
2. Deploy process with Task Listener of event type "complete" (e.g. [the one used for integration tests](https://github.com/camunda/camunda/blob/490565936577c887fc3df91a9f553ca1be087542/operate/qa/integration-tests/src/test/resources/task-listener.bpmn) ) and start an instance
3. Assign user task to yourself and complete it (it won't complete because the listener blocks it)
4. Fail the listener job that is created without retries left -> Incident should appear in Operate
5. Resolve incident via Operate UI (retry)
6. The retry count of the job should go up to 1 (from 0) and should complete normally

Hint: most of this is the same as for the implementation for Execution Listeners https://github.com/camunda/camunda/issues/23383#issuecomment-2459728833

- Versions to validate:
Operate: 8.7.0-alpha
Zeebe: 8.7.0-alpha

### :green_book: Link to the test case 

**PR**:
main: https://github.com/camunda/camunda/pull/24717

**Comment by vitorwtavares** [comment]
@kristinkomschow I was not very sure of what needed to be done here. If you could, please check if it's possible to improve and/or expand on the title and description so we can have better visibility into this specific issue.

**Comment by kristinkomschow** [comment]
@vitorwtavares Is there a separate issue/PR for e2e testing this feature? I'm asking because I'm unsure if I need to write manual e2e test instructions for this for QA.

**Comment by vitorwtavares** [comment]
> @vitorwtavares Is there a separate issue/PR for e2e testing this feature? I'm asking because I'm unsure if I need to write manual e2e test instructions for this for QA.

If you feel E2E testing should be done for this, it should be as part of the review process for this issue as I don't believe we have any other issues involving this, so I would say writing down instructions for the QA would be ideal.

**Comment by kristinkomschow** [comment]
Manual testing still needs to be done once the related zeebe PR was merged

**Comment by ce-dmelnych** [comment]
> [!NOTE]  
The related Zeebe PR (https://github.com/camunda/camunda/pull/26037) was merged yesterday, so this task should now be ready for E2E testing.

FYI: @kristinkomschow @ArezooAzar

**Comment by ArezooAzar** [comment]
@kristinkomschow : 

I’m trying to test the PR, but unfortunately, when running the Docker locally, I cannot see instances in Operate. I also attempted to test it in [SaaS](https://console.cloud.ultrawombat.com/org/6ff582aa-a62e-4a28-aac7-4d2224d8c58a/cluster/628b5f92-3186-4385-8c1f-429fc19b9b51), but the Tasklist is unhealthy. Since I need access to the Tasklist, I’m currently blocked until I can successfully run the environment.I already inform HTO team about the issue.

**Comment by ArezooAzar** [comment]
I tried to test this issue in SaaS, but it’s unclear how to run an execution listener there, and I haven’t been able to find help so far.
As a result, I’ll need to postpone testing until after the holidays. Hopefully, the issue with docker will be resolved by then, or someone can guide on executing a listener in SaaS.

**Comment by ArezooAzar** [comment]
@ce-dmelnych: @kristinkomschow  and   I tested the issue in [saas ](https://chaos-1.operate.ultrawombat.com/4a4bdc29-90b6-49e5-a1aa-fb294b456319/operate/processes/4503599627370507)and found that even though I set the retry count to 0 when make the job fails by using (zbctl fail job key --retries 0 --errorMessage "error message" ) , still the metadata shows 'retries left: 2', whereas the incident shows 'retries left: 0'.
<img width="1785" alt="Screenshot 2025-01-07 at 17 11 12" src="https://github.com/user-attachments/assets/479f2c12-e490-4db4-9371-b3ffa37d796f" />

@sdorokhova : can be because of exporter change ?

**Comment by kristinkomschow** [comment]
@ArezooAzar I tried this out locally today on the main branch, using `make env-up`, after changing the spring profiles that it uses to `-Dspring.profiles.active=operate,tasklist,broker,dev,auth` so I'd have tasklist available and no dev data.

I deployed a process with a "completing" task listener, triggered the completion via tasklist and then failed the job via zbctl. I got the incident and it also showed the retries left correctly, so I think this was probably related to one of the exporter issues that were still fixed recently.

<img width="686" alt="Image" src="https://github.com/user-attachments/assets/014eef48-a37d-410e-b369-4d537ed8ef9d" />

Could you maybe test this in SaaS one more time?

**Comment by kristinkomschow** [comment]
@ArezooAzar I think we should close this issue. The bug we found is very unlikely to have been introduced through the changes we made here. If want to still manually test this, we should ignore the retries count shown and instead just test if resolving the incident via the retry button works, so:
1. fail the listener task as described in the initial test setup
2. try to resolve the incident via Operate UI (this should activate the job again automatically)
3. complete the job via zbctl (or create a  job worker that does this automatically once the job is activated)

Zeebe does clearly actually fail the job with no retries, otherwise we wouldn't see the incident being created. So an attempt to complete the job would fail if Operate didn't (correctly) update the retry count (which is the behavior we originally wanted to test here).

This behavior also has automated test coverage by now, once this PR is merged: https://github.com/camunda/camunda/pull/27050

I've created an own issue for the bug of retries not being displayed correctly: https://github.com/camunda/camunda/issues/27899
